src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,910 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import {
+  getEmptyProfile,
+  getEmptyThread,
+} from '../../../profile-logic/data-structures';
+import BinaryPlistParser, { UID } from './BinaryPlistParser';
+import BinReader from './BinReader';
+import MaybeCompressedReader from './MaybeCompressedReader';
+
+//types
+import type {
+  Profile,
+  FuncTable,
+  FrameTable,
+  StackTable,
+} from '../../../types/profile';
+import type { UniqueStringArray } from '../../../../src/utils/unique-string-array';
+
+type InstrumentsSample = {|
+  +timestamp: number,
+  +threadID: number,
+  +backtraceID: number,
+  backtraceStack?: Array<number>,
+|};
+
+type TraceDirectoryTree = {|
+  +name: string,
+  +files: Map<string, File>,
+  +subdirectories: Map<string, TraceDirectoryTree>,
+|};
+
+type InstrumentsFrameInfo = {|
+  +key: string | number,
+  +name: string,
+  file?: string,
```

Comment:
I think it would be easier to make it a `string | null` and then you could make it readonly like other properties.

Reply:
Actually, I am writing this property in one place. If I do the change suggested by you, I am getting this error:
<img width="865" alt="Screenshot 2019-08-23 at 11 07 30 PM" src="https://user-images.githubusercontent.com/16744065/63612049-e8382400-c5fa-11e9-8328-60cc658ce8e4.png">

Reply:
If you make it `string | null` (and non optional), you could do:
```js
const frame: InstrumentsFrameInfo = {
  key: `${sourcePath}:${name}`,
  name: name,
  file: sourcePath || null,
};
```
And then you don't need to make it readwrite. Moreover it's easier to read at the location where you create the object.

Reply:
Made the change. thanks! @julienw